homeassistant/scripts/check_config.py

```diff
@@ -0,0 +1,172 @@
+import homeassistant.config as config_util
+# import homeassistant.util.yaml as yaml_util
+import homeassistant.loader as loader
+import homeassistant.util.yaml as yaml
+
+REQUIREMENTS = ['colorlog>2.1<3', 'colorama<=1']
+SHOW_FILES = True
+SHOW_SECRETS = True
+SHOW_FULL_CONFIG = True
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=too-many-locals,too-many-branches
+def run(script_args: List) -> int:
+    """Handle ensure config commandline script."""
+    # Color support
+    from colorama import init, Fore, Style
+    init()
+
+    def color(*args, c=Style.BRIGHT):  # pylint: disable=invalid-name
+        """Color helper."""
+        return c + ' '.join(args) + Style.RESET_ALL
+
+    parser = argparse.ArgumentParser(
+        description=("Check Home Assistant configuration."))
+    parser.add_argument(
+        '-c', '--config',
+        # metavar='path_to_config_dir',
+        default=config_util.get_default_config_dir(),
+        help="Directory that contains the Home Assistant configuration")
+    parser.add_argument(
+        '--script', choices=['check_config'])
+    parser.add_argument(
+        '--verbose', default=False)
+
+    args = parser.parse_args()
+
+    config_dir = os.path.join(os.getcwd(), args.config)
+    config_path = os.path.join(config_dir, 'configuration.yaml')
+    if not os.path.isfile(config_path):
+        print('Config does not exist:', config_path)
+        return 1
+
+    print(color("Testing configuration at", config_dir))
+
+    res = check(config_path)
+
+    if SHOW_FILES:  # Show Files
+        print(color('yaml files'), '(used /',
+              color('not used', c=Fore.RED)+')')
+        # Python 3.5 gets a recursive, but not in 3.4
+        for yfn in sorted(glob(os.path.join(config_dir, '*.yaml')) +
+                          glob(os.path.join(config_dir, '*/*.yaml'))):
+            the_color = '' if yfn in res['yaml_files'] else Fore.RED
+            print(color('-', yfn, c=the_color))
+
+    # pylint: disable=too-many-nested-blocks
+    if SHOW_FULL_CONFIG:
+        re_keys = re.compile(r"'([^']+)':")
+
+        def indent(item, indent0, indent1):
+            """Return indented text."""
+            return re_keys.sub(color(r'\1')+':',
+                               fill(pprint.pformat(item),
+                                    initial_indent=indent0,
+                                    subsequent_indent=indent1))
+
+        def printlist(obj, level=0, pref=''):
+            """Printlist."""
+            indent1 = level * ' '
+            indent0 = pref.rjust(level)
+            if isinstance(obj, list):
+                for ob0 in obj:
+                    printlist(ob0, level+2, '- ')
+            # elif isinstance(obj, dict):
+            #    for key0, val0 in obj.items():
+            #        printlist(val0, indent00 + key0 + ':' , indent1)
+            else:
+                print(indent(obj, indent0, indent1))
+
+        components = deep_convert_dict(res['components'])
```

Comment:
Why would you convert an ordereddict to a dictionary? OrderedDict is a child of dict, and thus has the same interface.